vault/provider.go

```diff
@@ -864,6 +865,48 @@ func ReadContextWrapper(f schema.ReadContextFunc) schema.ReadContextFunc {
 	}
 }
 
+func VersionCheckWrapper(f schema.CreateFunc, d *schema.ResourceDiff, meta interface{}, minVersion string) schema.CreateFunc {
+	return func(d *schema.ResourceData, meta interface{}) error {
+		client, e := provider.GetClient(d, meta)
+		if e != nil {
+			return e
+		}
+
+		featureEnabled, currentVersion, err := semver.SemanticVersionComparison(minVersion, client)
+		if err != nil {
+			return err
+		}
+
+		if !featureEnabled {
+			return fmt.Errorf("feature not enabled on current Vault version. min version required=%s "+
+				"current vault version=%s", minVersion, currentVersion)
+		}
+
+		return f(d, meta)
+	}
+}
+
+func VersionCheckContextWrapper(f schema.CreateContextFunc, minVersion string) schema.CreateContextFunc {
```

Comment:
Just another thought on naming:

```suggestion
func MinVersionCheckContextWrapper(f schema.CreateContextFunc, minVersion string) schema.CreateContextFunc {
```

Reply:
Updated in latest commit